jib-core/README.md

```diff
@@ -1,3 +1,116 @@
+
+## Upcoming features
+
+- Extensions to make building Java and other language-specific containers easier
+- Structured events to react to parts of Jib Core's execution
+
+See [Milestones](https://github.com/GoogleContainerTools/jib/milestones) for planned features. [Get involved with the community](https://github.com/GoogleContainerTools/jib/tree/master#get-involved-with-the-community) for the latest updates.
+
+## Adding Jib Core to your build
+
+Add Jib Core as a dependency using Maven:
+
+```xml
+<dependency>
+  <groupId>com.google.cloud.tools</groupId>
+  <artifactId>jib-core</artifactId>
+  <version>0.1.0</version>
+</dependency>
+```
+
+Add Jib Core as a dependency using Gradle:
+
+```groovy
+dependencies {
+  compile 'com.google.cloud.tools:jib-core:0.1.0'
+}
+```
+
+## Simple example
+
+```java
+Jib.from("busybox")
+   .addLayer(Arrays.asList(Paths.get("helloworld.sh")), AbsoluteUnixPath.get("/")) 
+   .setEntrypoint("/bin/sh", "-c", "chmod +x /helloworld.sh && /helloworld.sh")
+   .containerize(
+       Containerizer.to(RegistryImage.named("gcr.io/my-project/hello-from-jib")
+                                     .addCredential("myusername", "mypassword")));
+```
+
+1. [`Jib.from("busybox")`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/Jib.html#from-java.lang.String-) creates a new [`JibContainerBuilder`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/JibContainerBuilder.html) configured with [`busybox`](https://hub.docker.com/_/busybox/) as the base image.
+1. [`.addLayer(...)`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/JibContainerBuilder.html#addLayer-java.util.List-com.google.cloud.tools.jib.filesystem.AbsoluteUnixPath-) configures the `JibContainerBuilder` with a new layer with `helloworld.sh` (local file) to be placed into the container at `/helloworld.sh`.
+1. [`.setEntrypoint("/helloworld.sh")`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/JibContainerBuilder.html#setEntrypoint-java.lang.String...-) sets the entrypoint of the container to `/helloworld.sh`.
+1. [`RegistryImage.named("gcr.io/my-project/hello-from-jib")`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/RegistryImage.html#named-java.lang.String-) creates a new [`RegistryImage`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/RegistryImage.html) configured with `gcr.io/my-project/hello-from-jib` as the target image to push to.
+1. [`.addCredential`](http://static.javadoc.io/com.google.cloud.tools/jib-core/0.1.0/com/google/cloud/tools/jib/api/RegistryImage.html#addCredential-java.lang.String-java.lang.String-) adds the username/password credentials to authenticate the push to `gcr.io/my-project/hello-from-jib`.
```

Comment:
If you keep the `addCredential()` in the example, maybe note that it's optional as credentials are automatically resolved from the Docker configuration and credential helpers?

Reply:
Same as last comment - the credentials are only optional when using a registry that does not require credentials for push - otherwise, any automatically-resolved credentials must be fetched via a `CredentialRetriever`.